### PR TITLE
Create sns topics with or without subscriptions

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/ec2-app/tasks/sns.yml [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+
+- block:
+    - name: Create sns topics
+      sns_topic:
+        name: "{{ [item, opg_data.stack]|join('-') }}"
+        state: present
+        display_name: "{{ [item, opg_data.stack]|join('-') }}"
+      with_items: "{{ sns_topics|default([]) }}"
+      register: sns_topics_created
+  when: not sns_subscriptions|default(False)
```
Comment (carinadigital) [inline roles/ec2-app/tasks/sns.yml]: why not `sns_topics` condition?

Comment (brettminnie) [reply roles/ec2-app/tasks/sns.yml]: main.yml line 105
```
  - name: Create sns resources
      include: sns.yml
      when: sns_topics is defined
      tags:
        - sns
        - iam
```

File: roles/ec2-app/tasks/sns.yml [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
-- name: Create sns topics
-  sns_topic:
-    name: "{{ [item, opg_data.stack]|join('-') }}"
-    state: present
-    display_name: "{{ [item, opg_data.stack]|join('-') }}"
-  with_items: "{{ sns_topics|default([]) }}"
-  register: sns_topics_created
+
+- block:
+    - name: Create sns topics
```
Comment (carinadigital) [inline roles/ec2-app/tasks/sns.yml]: Better name please. something like `Create sns topics without default subscriptions`

Comment (brettminnie) [reply roles/ec2-app/tasks/sns.yml]: :+1:

**Review by carinadigital — CHANGES_REQUESTED** [review]
A bit or a worry that we have two sources of metadata for sns topics.

File: roles/ec2-app/tasks/sns.yml [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+- block:
+    - name: Create sns topics with default subscriptions
+      sns_topic:
+        name: "{{ [item.name, opg_data.stack]|join('-') }}"
+        state: present
+        display_name: "{{ [item.name, opg_data.stack]|join('-') }}"
+        subscriptions: "{{ item.subscriptions }}"
+      with_items: "{{ sns_subscriptions|default([]) }}"
+      register: sns_topics_created
+  when: sns_subscriptions|default(False)
```
Comment (carinadigital) [inline roles/ec2-app/tasks/sns.yml]: can this variable be better named to show what it is? e.g. `sns_topics_with_subscriptions`

**Review by carinadigital — APPROVED** [review]

**Pull request merged**